consensus/core/src/network/tonic_network.rs

```diff
@@ -144,15 +145,18 @@ impl NetworkClient for TonicClient {
         let response = client.subscribe_blocks(request).await.map_err(|e| {
             ConsensusError::NetworkRequest(format!("subscribe_blocks failed: {e:?}"))
         })?;
-        let stream = response.into_inner().filter_map(move |b| async move {
-            match b {
-                Ok(response) => Some(response.block),
-                Err(e) => {
-                    debug!("Network error received from {}: {e:?}", peer);
-                    None
+        let stream = response
+            .into_inner()
+            .take_while(|b| futures::future::ready(b.is_ok()))
```

Comment:
How is this line useful?

Reply:
Its to temporarily work around a small bug i found in tonic that i've fixed upstream but there isn't a release out for it yet: https://github.com/hyperium/tonic/pull/1823